### PR TITLE
Fixes and Updates

### PR DESCRIPTION
**Pull request by mblunt**
Hello!

My name is Michael, I am the current HackPack president. Took the day to work on some club stuff, and one of those things was working on this bot. It had been crashing, looks like the discord API has changed since this was initially written. Changes are just bug fixes, nothing major and no feature additions.

### CODE REVIEW

**Comment by weakphish** [comment]
hey, thanks for taking the time to do this! if you could, you can go ahead and just remove the `commands.go.old` file - not neccesary to keep it around if it doesn't work.

Thanks!

**Comment by weakphish** [comment]
also, i'll go ahead and give you repo access

File: commands.go [hunk 1]
```diff
@@ -1,329 +1,352 @@
...
+				{
+					Name:        "join",
+					Type:        discordgo.ApplicationCommandOptionSubCommand,
+					Description: "Join a CTF",
+					Options: []*discordgo.ApplicationCommandOption{
+						{
+							Type:        discordgo.ApplicationCommandOptionString,
+							Name:        "ctf-name",
+							Description: "CTF name",
+							Required:    false,
```
Comment (quantatic) [inline commands.go]: Should this be required?

File: commands.go [hunk 2]
```diff
@@ -1,329 +1,352 @@
...
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Create a map of <CommandName>:<HandlerFunction> for each command. Each command will
+	// correspond to a first-class function that will handle the command's usage upon invocation
+	commandHandlers = map[string]func(s *discordgo.Session, i *discordgo.InteractionCreate){
```
Comment (quantatic) [inline commands.go]: Are these used anywhere?

**Review by quantatic — APPROVED** [review]

File: commands.go (hunk 1 above)
Comment (weakphish) [reply commands.go]: agree w/ @quantatic  here, probably make this required

File: README.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
 # hackpack-bot
 NC State's Hackpack Discord Bot, written in Go.
-Maintained by John Allison (john123allison@gmail.com)
+Created by John Allison (john123allison@gmail.com)
+
+Maintained by Michael Blunt (mbblunt@ncsu.edu)
```
Comment (weakphish) [inline README.md:5]: also, while it is accurate that I will/am handing over maintenance to the active HackPack team, it's generally good practice to ask directly before making a change like this. No harm no foul, just a heads up.

Comment (mblunt) [reply README.md:5]: Sorry! Didn't mean anything bad by that, thanks for letting me know. This is my first PR, still a little rough on the edges with the etiquette :)

**Comment by mblunt** [comment]
Alright, I made the above changes 😄. Just wanted to check this is good to merge?

File: README.md (hunk 3 above)
Comment (weakphish) [reply README.md:5]: All good, don't sweat it ! :)

**Comment by weakphish** [comment]
> Alright, I made the above changes 😄. Just wanted to check this is good to merge?

merge away, mr. president 🫡

**Pull request merged**